### PR TITLE
[Ray] RayExecutionContext.get_chunk_meta from meta service

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?

To make the auto merge chunk feature works, Ray context should get meta from meta service if the key does not exist in current context.

## Related issue number

Fixes https://github.com/mars-project/mars/issues/3211
Fixes https://github.com/mars-project/mars/issues/3220

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: ci/reload-env.sh [hunk 1]
```diff
@@ -91,3 +91,5 @@ function retry {
 alias pip="retry pip"
 shopt -s expand_aliases
 set -e
+# https://github.com/pypa/setuptools/issues/3504
+export SETUPTOOLS_ENABLE_FEATURES="legacy-editable"
```
Comment (chaokunyang) [inline ci/reload-env.sh:95]: This should be reverted, setuptools has been fixed in master

Comment (fyrestone) [reply ci/reload-env.sh]: Thanks. I removed this commit and merged latest master.

**Review by chaokunyang — DISMISSED** [review]
LGTM

File: mars/tests/test_session.py [hunk 2]
```diff
@@ -519,11 +519,11 @@ def test_merge_groupby(setup, method, auto_merge):
     rs = np.random.RandomState(0)
     raw1 = pd.DataFrame({"a": rs.randint(3, size=100), "b": rs.rand(100)})
     raw2 = pd.DataFrame({"a": rs.randint(3, size=10), "c": rs.rand(10)})
-    df1 = md.DataFrame(raw1, chunk_size=10)
-    df2 = md.DataFrame(raw2, chunk_size=10)
+    df1 = md.DataFrame(raw1, chunk_size=10).execute()
+    df2 = md.DataFrame(raw2, chunk_size=10).execute()
```
[the remark this answers is not in the document]
Comment (fyrestone) [reply mars/tests/test_session.py:523]: Yes, this is to test the corner case of get chunk meta. If `execute()`, then the df1, df2 will be in a dedicated mars task. The following execution will be a new task, the new task needs to get chunk meta across stages / tasks to do the auto merging.

File: LICENSE [hunk 3]
```diff
@@ -127,7 +127,7 @@ Copyright 1999-2021 Alibaba Group Holding Ltd.
       may provide additional or different license terms and conditions
       for use, reproduction, or distribution of Your modifications, or
       for any such Derivative Works as a whole, provided Your use,
-      reproduction, and distribution of the Work otherwise complies with
+      reproduction, and distribution of the Work otherwise compiles with
```
[the remark this answers is not in the document]
Comment (fyrestone) [reply LICENSE:130]: The spelling check may be updated, so some typo can be catched by CI.

**Review by zhongchun — APPROVED** [review]
LGTM.

**Review by chaokunyang — APPROVED** [review]
LGTM
